app/controllers/legacy/datasets_controller.rb

```diff
@@ -1,6 +1,20 @@
 class Legacy::DatasetsController < ApplicationController
+  rescue_from Dataset::DatasetNotFound, with: :available_soon
+
   def redirect
     dataset = Dataset.get_by_legacy_name(legacy_name: params[:legacy_name])
     redirect_to dataset_path(dataset.uuid, dataset.name), status: :moved_permanently
   end
+
+  def available_soon
+    if request_originates_from_legacy?
+      render 'datasets/available_soon'
+    else
+      raise
+    end
+  end
+
+  def request_originates_from_legacy?
+    request.referrer.present? && URI.parse(request.referrer).host == ENV['LEGACY_HOST']
```

Comment:
Find and legacy will share the same host. I think we'll need to look at the path of the referrer.